services/order.py

```diff
@@ -0,0 +1,34 @@
+from django.contrib.auth import get_user_model
+from django.db.models import QuerySet
+from django.db import transaction
+
+
+from db.models import Ticket, Order
+
+
+@transaction.atomic
+def create_order(tickets: list[dict],
+                 username: str,
+                 date: str = None
+                 ) -> QuerySet:
+    user = get_user_model().objects.get(username=username)
+    order = Order.objects.create(user=user)
+
+    if date:
+        order.created_at = date
+
+    for ticket in tickets:
+        Ticket.objects.create(order=order,
+                              movie_session_id=ticket["movie_session"],
+                              row=ticket["row"],
+                              seat=ticket["seat"]
+                              )
+    order.save()

```

Comment:
Don't Repeat Yourself: You are calling the `save()` method even if the `date` is not provided. It's better to call `save()` only if the `date` is provided, to avoid unnecessary database operations. You can use the `update_fields` argument of the `save()` method to save only the `created_at` field. Like this:

python
if date:
    order.created_at = date
    order.save(update_fields=['created_at'])
```